src/Sarif.Driver/Sdk/AnalyzeCommandBase.cs

```diff
@@ -168,7 +170,30 @@ private bool ValidateFile(TContext context, string filePath, bool? shouldExist)
             return false;
         }
 
-        internal AggregatingLogger InitializeLogger(IAnalyzeOptions analyzeOptions)
+        private static bool ValidateInvocationPropertiesToLog(TContext context, IEnumerable<string> propertiesToLog)
+        {
+            bool succeeded = true;
+
+            if (propertiesToLog != null)
+            {
+                List<string> validPropertyNames = typeof(Invocation).GetProperties(BindingFlags.Public | BindingFlags.Instance)
```

Comment:
You could have used a dictionary here initialized with a case-insensitive comparer, and skipped all the ToUpper calls. Just create the enumerable list as you're doing and pass that to the Dictionary ctor.